File: app/controllers/UserController.scala
package controllers

import javax.inject.Inject
import models._
import play.api.data._
import play.api.data.Forms._
import play.api.mvc._

class UserController @Inject()(cc: ControllerComponents) extends AbstractController(cc) with play.api.i18n.I18nSupport {
  /* Formを定義 */
  val userForm = Form(
    // モデル User とマッピング
    // mapping ( "カラム名" -> 型名 )
    mapping(
      "account" -> nonEmptyText,
      "email" -> nonEmptyText,
      "kenpos_id" -> nonEmptyText,
      "kenpos_passwd" -> nonEmptyText)(User.apply)(User.unapply)
  )

  /* テンポラリー表示用 */
  def entryInit = Action {
    implicit request =>
      // Form.fill：既存の値で埋める。フォームを編集する用途で使用する
      val filledForm = userForm.fill(User("Taro_Kenpos1", "Taro@Kenpos.co.jp", "sss012345", "*****"))
      Ok(views.html.tags.entry(filledForm))
  }

  /* 登録用関数 */
  def entrySubmit = Action {
    implicit request =>
      // bindFromRequest.get or fold でリクエストからフォーム情報を取得。エラー処理を行えるfoldを推奨。getの場合は取得に失敗すると例外をスロー
      userForm.bindFromRequest.fold(
        formWithErrors => {
          println("entrySubmit:Error" + formWithErrors)
          BadRequest(views.html.tags.entry(formWithErrors))
        },
        userData => {
          println(userData)
          Ok(views.html.tags.enrtySubmit())
        }
      )
  }
}

File: app/controllers/UserController.java
package controllers;


import models.Login;
import models.UserModel;
import play.data.Form;
import play.mvc.Controller;
import play.mvc.Result;
import views.html.template.create;

import java.security.NoSuchAlgorithmException;

public class UserController extends Controller {

    /**
     * ユーザ作成(データ入力)
     *
     * @return
     */
    public Result add() {
        Form<UserModel> f = new Form<UserModel>(UserModel.class);
        return ok(create.render("入力フォーム", f));
    }

    /**
     * ユーザ作成(データベースに反映させる)
     *
     * @return
     */
    public Result create() throws NoSuchAlgorithmException {
        Form<UserModel> f = new Form<UserModel>(UserModel.class).bindFromRequest();
        if (!f.hasErrors()) {
            UserModel data = f.get();
            Login hashedPassword= new Login();
            String password = data.getPassword();
            data.setPassword(hashedPassword.sha512(password));
            try{
                data.save();
            } catch (Exception e){
                System.out.println(e);

            }
            System.out.println(data.getUsername());
            return redirect("/");
        } else {
            return badRequest(create.render("ERROR", f));
        }
    }

    /**
     * ユーザ登録フォーム管理クラス
     */
    public static class userForm {
        public String input;
        public String userid;
        public String password;
        public String nickname;
        public String email;

    }
}
